### PR TITLE
fix(rum-core): handle empty marks when compressing (#1104)

### PR DESCRIPTION
**Pull request by devcorpio**
# Summary

In order to prevent the agent to throw an error when compressing transactions we are now checking the existence of navigation timing and agent marks.

Following the scout principle news tests added which check that marks are being compressed properly 👍

### CODE REVIEW

File: packages/rum-core/src/common/compress.js [hunk 1]
```diff
@@ -120,26 +120,56 @@ function compressMarks(marks) {
...
+      compressed.fp = fp
+    }
+    if (lp) {
+      compressed.lp = lp
+    }
   }
-  if (lp) {
-    compressed.a.lp = lp
+
+  if (Object.keys(compressed).length === 0) {
```
Comment (vigneshshanmugam) [inline packages/rum-core/src/common/compress.js:171]: nit: we dont need this check, truncate model should take care of properties without keys.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

File: packages/rum-core/src/common/compress.js (hunk 1 above)
Comment (devcorpio) [reply packages/rum-core/src/common/compress.js]: I removed the check since the intake api is accepting an empty object as well.

Comment (vigneshshanmugam) [reply packages/rum-core/src/common/compress.js:171]: Ah good point, null is better as as fields with null values will be removed when serialized,

**Pull request merged**